bench/workgen/workgen.cpp

```diff
@@ -523,9 +545,17 @@ WorkloadRunner::start_tables_create(WT_CONNECTION *conn)
         const std::string config(
           "key_format=S,value_format=S,app_metadata=\"" + DYN_TABLE_APP_METADATA);
 
+        uint32_t num_files = get_dir_num_files(_wt_home);
         int creates = 0, retries = 0;
-        while (
-          !stopping && creates < _workload->options.create_count && retries < TABLE_MAX_RETRIES) {
+        // Make sure not to exceed the maximum number of files that can exist in the database.
+        int num_files_to_create = 0;
+        if (num_files < max_files) {
+            if (num_files + _workload->options.create_count > max_files)
+                num_files_to_create = max_files - num_files;
+            else
+                num_files_to_create = _workload->options.create_count;
```

Comment:
We should assert that the `create_count` has to be lower than the max_files variable.

Reply:
Added a check in 3889e778b01f266914e8ed89d16646e79b535e99.